src/screens/wallets/wallets.tsx

```diff
@@ -120,26 +136,101 @@ export class WalletsScreenComponent extends React.Component<IProps & IReduxProps
         );
     }
 
+    public onPressDelete(wallet: IWalletState) {
+        // show a confirm dialog
+        Alert.alert(translate('Wallets.deleteWallet'), translate('Wallets.confirmDelete'), [
+            {
+                text: translate('App.labels.cancel'),
+                onPress: () => {
+                    /* console.log('Cancel Pressed')*/
+                },
+                style: 'cancel'
+            },
+            {
+                text: translate('App.labels.delete'),
+                onPress: () => {
+                    this.onDeleteConfirmed(wallet);
+                }
+            }
+        ]);
+    }
+
+    public onDeleteConfirmed(wallet: IWalletState) {
+        // request password
+        this.setState({
+            showPasswordModal: true
+        });
+
+        this.passwordModalRef.requestPassword(async password => {
+            // verify password by decrypting mnemonic (?)
+            try {
+                const passHash = await hash(password);
```

Comment:
This is ok for HDWallets, but when we introduce Hardware wallets this code will not work, since we won't have the mnemonic for the wallet.  We can leave it as it is for now, but we need to find a solution to check user password.

Reply:
changed to use pass validation in pass modal